### PR TITLE
add factory contract and draft of process

### PR DESCRIPTION
**Pull request by shestakoven**

### CODE REVIEW

File: contracts/rapidMundaFactory.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+contract ProcessDefinition {
+    // draft
+    address public factory;
+    constructor() {
+        factory = msg.sender;
+    }
+}
+
+contract RapidMundaFactory {
```
Comment (evahteev) [inline contracts/rapidMundaFactory.sol]: Chainflow

**Review by evahteev — COMMENTED** [review]
1. Events needs to be emmited, like:
    event ProcessDefinitionDeployed(bytes32 indexed bpmnHash, address processAddress);
    event ProcessInstanceStarted(string processInstanceID);
    event ProcessInstanceFinished(string processInstanceID);

2. Method to work with process instances start/end and status reports
3. Process definition and Process instance should have owners

File: contracts/rapidMundaFactory.sol (hunk 1 above)
Comment (shestakoven) [reply contracts/rapidMundaFactory.sol:14]: Should be the ProcessInstance separate contract, or we'll just keep an info about instance in the ProcessDefinition?

**Review by kelar86 — APPROVED** [review]

File: contracts/ProcessDefinition.sol [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    mapping(string => address) public processInstanceOwners;
+    // List of all started process instance IDs.
+    string[] private list;
+
+    address public owner;
+    uint public processPrice;
+
+    address public feeClaimer;
+    // 5% fee
+    uint public fee = 5;
```
Comment (evahteev) [inline contracts/ProcessDefinition.sol:20]: should be param on deploy

File: contracts/ProcessDefinition.sol [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        emit ChangeFeeClaimer(_feeClaimer);
+    }
+
+    function transferOwnership(address _newOwner) external onlyOwner {
+        owner = _newOwner;
+        emit TransferOwnership(_newOwner);
+    }
+
+    // Starts a new process instance.
+    function start(string memory processInstanceId) external payable {
```
Comment (evahteev) [inline contracts/ProcessDefinition.sol:70]: please create end method, which would substract fee + process price from locked tokens.

Comment (shestakoven) [reply contracts/ProcessDefinition.sol:70]: fee transfer happens on the start event (see below). It's not as optimal for gas, but it's easier to handle. As for the withdraw for the process owner, we have an "withdraw" function which he can call at any time and get his tokens with the fee deducted.

So do we need the "end" function?

**Review by evahteev — APPROVED** [review]

**Pull request merged**